src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -827,6 +848,10 @@ private Object readResolve() {
             storageAccountType = SkuName.STANDARD_LRS.toString();
         }
 
+        if (nodeProperties == null) {
+            nodeProperties = new DescribableList<>(Saveable.NOOP);
```

Comment:
I noticed a null pointer when it was being deserialised and no node properties had been set, this will work around that